cwl/src/main/scala/cwl/CommandOutputExpression.scala

```diff
@@ -0,0 +1,128 @@
+import scala.concurrent.duration._
+
+case class CommandOutputExpression(outputBinding: CommandOutputBinding,
+                                   override val cwlExpressionType: WomType,
+                                   override val inputs: Set[String]) extends CwlWomExpression {
+
+  // TODO WOM: outputBinding.toString is probably not be the best representation of the outputBinding
+  override def sourceString = outputBinding.toString
+
+
+
+  override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[WomValue] = {
+
+    val parameterContext = ParameterContext().addInputs(inputValues)
+
+    /*
+    CommandOutputBinding.glob:
+    Find files relative to the output directory, using POSIX glob(3) pathname matching. If an array is provided, find
+    files that match any pattern in the array. If an expression is provided, the expression must return a string or an
+    array of strings, which will then be evaluated as one or more glob patterns. Must only match and return files which
+    actually exist.
+
+    http://www.commonwl.org/v1.0/CommandLineTool.html#CommandOutputBinding
+     */
+    def commandOutputBindingToWomValue: Either[NonEmptyList[String], WomValue] = {
+      import StringOrExpression._
+      (outputBinding, parameterContext) match {
+        case (CommandOutputBinding(_, _, Some(String(value))), Right(_)) => WomString(value).asRight
+        case (CommandOutputBinding(Some(glob), _, None), Right(parameterContext)) =>
+              WomArray(WomArrayType(WomStringType), GlobEvaluator.globPaths(glob, parameterContext, ioFunctionSet).map(WomString.apply)).asRight
+
+        case (CommandOutputBinding(glob, loadContents, Some(Expression(expression))), Right(parameterContext)) =>
+
+          val paths: Seq[String] = glob.toSeq flatMap { globValue =>
+            GlobEvaluator.globPaths(globValue, parameterContext, ioFunctionSet)
+          }
+
+          val _loadContents: Boolean = loadContents getOrElse false
+
+          val womMaps: Array[Map[String, String]] =
+            paths.map({
+              (path:String) =>
+                // TODO: WOM: basename/dirname/size/checksum/etc.
+
+                val contents: Map[String, String] =
+                    Map("contents" -> load64KiB(path, ioFunctionSet)).filter(_ => _loadContents)
+
+                Map(
+                  "location" -> path
+                ) ++ contents
+            }).toArray
+
+          val outputEvalParameterContext: ParameterContext = parameterContext.setSelf(womMaps)
+
+          expression.fold(EvaluateExpression).apply(outputEvalParameterContext).toEither.leftMap(e => NonEmptyList.one(e.getMessage))
+      }
+    }
+    //To facilitate ECMAScript evaluation, filenames are stored in a map under the key "location"
+    val womValue =
+      commandOutputBindingToWomValue map {
+        case WomArray(_, Seq(WomMap(WomMapType(WomStringType, WomStringType), map))) => map(WomString("location"))
```

Comment:
ToL [Even Dora](https://youtu.be/gRup3mJCmm4?t=6s) thinks this is a lot of `map`. Perhaps a different name for the pattern var and/or the use of `apply`.